### ISSUE TITLE
CMIP6 cmor table error ssp370

### ISSUE THREAD

**Issue by ledm** [ESMValGroup/ESMValCore#284]
There's an error in the CMOR table for the experiment SSP3 7.0. 

Looking at the file [`esmvalcore/cmor/tables/cmip6/Tables/CMIP6_CV.json`, line 8414](https://github.com/ESMValGroup/ESMValCore/blob/8a762f026e4df44929f8f0503bbc35251d90a55b/esmvalcore/cmor/tables/cmip6/Tables/CMIP6_CV.json#L8414):
```
            "ssp370":{
                "activity_id":[
                    "ScenarioMIP AerChemMIP"
                ],
```

When ESMValCore looks for the ssp370 experiemnt on BADC, it tries to set the `[activity]` section to `ScenarioMIP AerChemMIP`, which of course doesn't exist. 

The error message is: 
```
2019-10-01 12:46:05,861 UTC [62192] DEBUG   Skipping non-existent
 /badc/cmip6/data/CMIP6/ScenarioMIP AerChemMIP/MOHC/UKESM1-0-LL
/ssp370/r1i1p1f2/Amon/tas/gn/[latestversion]
```

I've made a local fix for this, but I don't think that we're supposed to edit the CMOR tables. Do we need to raise it with them? Alternatively, should we instead fix `_recipe.py` so that it can deal with spaces in the CMOR table entries?

**Comment by valeriupredoi** [comment]
this should definitely be raised with the CMOR folk since the `activity_id` is a list of mip's and this is clearly a malformation; we can put a check and split the string by space, but this will mask any other possible issues

**Comment by ledm** [comment]
Created CMOR issue here: https://github.com/PCMDI/cmip6-cmor-tables/issues/264

**Comment by taylor13** [comment]
@ledm @ValerioLembo:  According to the first entry in the first table in https://goo.gl/v1drZl, which defines the global attributes, ``activity_id`` should be a list of space-separated activities, so I think CMIP6_CV.json is correct.  Do you agree?

**Comment by ledm** [comment]
Hi @taylor13, thanks for the quick reply. That is indeed correct. It looks like the problem is at our end.

@valeriupredoi, @matti, @bouweandela, I guess we need to change `_recipy.py` such that it can deal with space characters in `activity`. However, looking at @taylor13's document, we also need to mange spaces in the following CMIP data attributes:
- `activity`
- `external_variables`
- `realm`
- `source`
Are we handling this metadata correctly at the moment?

**Comment by valeriupredoi** [comment]
did you want to tag @mattiarighi instead? I also posted a comment on the original CMOR issue since it is slightly confusing to as to which is the accepted syntax :beer:

**Comment by ledm** [comment]
Yes, but in all cases, they are adheiring to their own rules. We need to adapt ESMValTool to be able to copy with space separated activities in the CMOR tables. How do we do that?

**Comment by valeriupredoi** [comment]
@jvegasbsc is the CMOR guru :grin:

**Comment by valeriupredoi** [comment]
also if the guru is siasta-ing, in `cmor/table.py` l.194:
```
            with open(json_file) as inf:
                table_data = json.loads(inf.read())
                try:
                    exps = table_data['CV']['experiment_id']
                    for exp_id in exps:
                        activity = exps[exp_id]['activity_id'][0].split(' ')
                        self.activities[exp_id] = activity
                except (KeyError, AttributeError):
                    pass
```
so we need to put a check see if the `activity` is a list

**Comment by zklaus** [comment]
This was specifically addressed in #248 and resolved in #257. I can confirm that it works for me exactly with ssp370. @ledm, could you please double check that your esmvalcore is up-to-date and still produces the error?
